doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+```bash
+config ntp vrf default
+config ntp vrf mgmt
+```
+
+**The following command adds/deletes NTP server:**
+```bash
+config ntp server add 0.ua.pool.ntp.org \
+  --type server \
+  --key 42 \
+  --version 4
+config ntp server del 0.ua.pool.ntp.org
+```
+
+**The following command enabled/disables specific NTP server:**
+```bash
+config ntp server 0.ua.pool.ntp.org state enabled
+config ntp server 0.ua.pool.ntp.org state disabled
+```
+
+**The following command adds/removes NTP authentication key:**
+```bash
+config ntp key add 42 theanswer \
+  --type sha1 \
+  --trusted yes
+config ntp key del 42
+```
+
+#### 2.6.2.2 Show command group
+
+**The following command shows NTP state:**
+```
+root@sonic:/home/admin# show ntp
+MGMT_VRF_CONFIG is not present.
+synchronised to NTP server (10.10.10.2) at stratum 5
+   time correct to within 287 ms
+   polling server every 64 s
+     remote           refid      st t when poll reach   delay   offset  jitter
+==============================================================================
++10.10.10.1     162.159.200.123  4 u   53   64  377    0.134  +240.45  10.868
+*10.10.10.2     162.159.200.123  4 u   43   64  377    0.110  +231.47   5.604
+```
+
+**The following command shows NTP global config:**
+```
+root@sonic:/home/admin# show ntp global
+State: enabled
+Authentication: enabled
+VRF: default
+```
+
+**The following command shows NTP servers:**
+```
+root@sonic:/home/admin# show ntp server
+SERVER               STATE     TYPE    KEY  VERSION
+-------------------  --------  ------  ---  -------
+time.cloudflare.com  enabled   server       4
+2001:4860:4806::     enabled   server       3
+216.239.35.4         disabled  server  42   4
+pool.ntp.org         disabled  pool         4
+```
+
+**The following command shows NTP keys:**
+```
+root@sonic:/home/admin# show ntp key
+KEY ID  TYPE  VALUE       TRUSTED
+------  ----  ----------  -------
+1       md5   mypass      yes
+2       sha1  thebestntp  no
+42      md5   theanswer   yes
+```
+
+## 2.7 YANG model
+An existing YANG model `sonic-ntp.yang` will be extended in order to provide support for configuring global NTP and NTP servers.
+
+**Skeleton code:**
+```yang
+module sonic-ntp {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-ntp";
+    prefix ntp;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-mgmt_vrf {
+        prefix mvrf;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    import sonic-loopback-interface {
+        prefix lo;
+    }
+
+    import sonic-mgmt_port {
+        prefix mprt;
+    }
+
+    description
+        "NTP yang Module for SONiC OS";
+
+    revision 2021-04-07 {
+        description
+            "First revision";
+    }
+
+    typedef state {
+        description "State of the config";
+        type enumeration {
+            enum enabled;
+            enum disabled;
```

Comment:
These typedefs should be moved to sonic types YANG and use the existing typedefs as much as possible.

Reply:
ok